velox/functions/sparksql/Arithmetic.h

```diff
@@ -318,4 +318,62 @@ struct IsNanFunction {
     }
   }
 };
+
+FOLLY_ALWAYS_INLINE static int8_t fromHex(char c) {
```

Comment:
Could we reuse the existing function to avoid duplication?
https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/BinaryFunctions.h#L256

Reply:
maybe not, these two function's behavior is different.

Reply:
Is the difference about throwing exception or returning invalid ouput for a failure? Maybe [Status](https://github.com/oap-project/velox/blob/update/velox/common/base/Status.h) could be used as function output to represent success or error, and it could be handled in the caller.

Reply:
@Yohahaha Would you reply to @rui-mo 's comment?

Reply:
```
  if (c >= '0' && c <= '9') {
    return c - '0';
  }

  if (c >= 'A' && c <= 'F') {
    return 10 + c - 'A';
  }

  if (c >= 'a' && c <= 'f') {
    return 10 + c - 'a';
  }
```

We can only reuse few codes above, Spark's implementation has lots of differences compare with Velox's (see PR description), mixed two implementations into one common function bring more complexity and hurt readability from my view. However, if you insist on extract these codes into one function, I will try later. @mbasmanova @rui-mo